src/Access/ContactFormsAccessCheck.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_contact_forms\Access;
+
+use Drupal\contact\ContactFormInterface;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Access\AccessResultInterface;
+use Symfony\Component\Routing\Route;
+
+/**
+ * Route access handler for contact form routes.
+ */
+class ContactFormsAccessCheck implements AccessInterface {
+
+  /**
+   * Denies access to the corporate contact form if not meant for canonical URL.
+   *
+   * @param \Symfony\Component\Routing\Route $route
+   *   The route.
+   * @param Drupal\contact\ContactFormInterface $contact_form
+   *   The contact form to check against.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   The access result.
+   */
+  public function access(Route $route, ContactFormInterface $contact_form = NULL): AccessResultInterface {
+    $cache = new CacheableMetadata();
+    $cache->addCacheableDependency($contact_form);
+    $cache->addCacheableDependency($route);
```

Comment:
Route is a symfont object. It doesnt have cache metadata. You have to add the route cache context here...

Reply:
I'm thinking route.name ?